### DISCUSSION TITLE
[boost-static-string] Standalone mode

### DISCUSSION THREAD

**Discussion by sylveon** [microsoft/vcpkg#23003] [Q&A]
****

**Is your feature request related to a problem? Please describe.**
Boost.StaticString supports a standalone mode that allows the library to be used with no Boost dependency when the compiler uses C++17 or higher. Currently, installing boost-static-string with vcpkg always brings a ton of Boost related packages even if I only intend to use it in standalone mode.

**Proposed solution**
There should be a feature package for standalone mode. Something like `vcpkg install boost-static-string[standalone]`.

**Describe alternatives you've considered**
Submoduling Boost.StaticString, but that defeats the point of using a package manager like vcpkg in the first place.

**Additional context**
![image](https://user-images.githubusercontent.com/6440374/152664586-6ef6ed7f-5ae5-4a96-a0c0-6c024584b492.png)

**Comment by PhoebeHui** [comment]
It looks break our [feature policy](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#features) please try [overlay ports](https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md) instead.

**Comment by sylveon** [comment]
I'm not exactly sure which part of the feature policy it breaks.

**Reply by mathisloge** [reply]
features should be additive. so if the `standalone` "feature" would alter the boost API in a way in which a dependency which don't need the `standalone` feature would break, it can't be a feature.

In other words: libraries which depend on boost-static-string shouldn't break if the `standalone` feature is on. If any library can't be build,it is either a bug in the library or a case for the overlay ports


In case of `boost-static-string[standalone]` it would break because the "normal" installation brings all those other boost deps in and the standalone feature would remove them.
